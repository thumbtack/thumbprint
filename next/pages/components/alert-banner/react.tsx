import React from 'react';
import type { InferGetStaticPropsType } from 'next';
import { AlertBanner } from '@thumbtack/thumbprint-react';
import { MDXComponentPage } from '../../../components/mdx/mdx';
import getComponentPageStaticProps from '../../../utils/get-component-page-static-props';
import { CodeExperimental, H2, H3 } from '../../../components/mdx/components';

export const getStaticProps = () => {
    return getComponentPageStaticProps({
        title: 'Alert Banner',
        description: 'Important account information displayed at the top of a screen',
        componentId: 'alert-banner',
        platformId: 'react',
    });
};

export default function ComponentDocumentationPage({
    layoutProps,
    componentPageProps,
}: InferGetStaticPropsType<typeof getStaticProps>) {
    return (
        <MDXComponentPage componentPageProps={componentPageProps} layoutProps={layoutProps}>
            <H2>Themes</H2>

            <H3>Caution</H3>

            <CodeExperimental language="jsx">
                <AlertBanner theme="caution">
                    Your pre-paid credits are running low. Purchase new credits before you run out.{' '}
                    <a href="https://example.com/">Purchase credits</a>
                </AlertBanner>
            </CodeExperimental>

            <H3>Info</H3>

            <CodeExperimental language="jsx">
                <AlertBanner theme="info">
                    You havenâ€™t finish setting up. Add 3 reviews and turn on targeting.{' '}
                    <a href="https://example.com/">Finish setup</a>
                </AlertBanner>
            </CodeExperimental>

            <H3>Warning</H3>

            <CodeExperimental language="jsx">
                <AlertBanner theme="warning">
                    Oops. Your payment failed! Update your payment details to continue using
                    Thumbtack. <a href="https://example.com/">Update payment</a>
                </AlertBanner>
            </CodeExperimental>
        </MDXComponentPage>
    );
}
