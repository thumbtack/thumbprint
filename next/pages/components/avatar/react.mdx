import { MDXComponentPage } from '../../../components/mdx/mdx';
import baseGetStaticProps from '../../../utils/mdx-get-static-props';
export const getStaticProps = ctx => baseGetStaticProps(ctx, metadata);
export default props => <MDXComponentPage {...props} {...metadata} />;

export const metadata = {
    title: 'Avatar',
    description: 'Display user images and badges on Thumbtack.',
    component: {
        id: 'avatar',
        platformId: 'react',
    },
};

import { UserAvatar, EntityAvatar } from '@thumbtack/thumbprint-react';
import DeprecatedComponentAlert from '../../../components/thumbprint-components/deprecated-component-alert/deprecated-component-alert.tsx';

## Avatar sizes

`UserAvatar`is available in five sizes ranging from `xlarge` to `xsmall`.

```jsx
<>
    <div className="flex items-center mb4">
        <div className="mr3">
            <UserAvatar imageUrl="https://i.pravatar.cc/140" size="xlarge" />
        </div>
        <div className="mr3">
            <UserAvatar imageUrl="https://i.pravatar.cc/140" size="large" />
        </div>
        <div className="mr3">
            <UserAvatar imageUrl="https://i.pravatar.cc/140" size="medium" />
        </div>
        <div className="mr3">
            <UserAvatar imageUrl="https://i.pravatar.cc/140" size="small" />
        </div>
        <div className="mr3">
            <UserAvatar imageUrl="https://i.pravatar.cc/140" size="xsmall" />
        </div>
    </div>
</>
```

## Avatars without images

Avatars without images can display the the user or entityâ€™s initials instead. The initials are assigned to a color based on the first letter in the `initials` prop.

```jsx
<>
    <div className="flex mb4">
        <div className="mr2">
            <UserAvatar initials="AA" size="large" />
        </div>
        <div className="mr2">
            <UserAvatar initials="BA" size="large" />
        </div>
        <div className="mr2">
            <UserAvatar initials="CA" size="large" />
        </div>
        <div className="mr2">
            <UserAvatar initials="DA" size="large" />
        </div>
        <div className="mr2">
            <UserAvatar initials="EA" size="large" />
        </div>
        <div>
            <UserAvatar initials="HA" size="large" />
        </div>
    </div>
</>
```

## Badging

A badge can be added to denote the online status of a user. The badge will be placed on the top right portion of the avatar and use the 400-level green color as positive indicator of their online status.

### Online

This badge indicates that a user or entity is online.

```jsx
<div>
    <div className="flex items-center mb4">
        <div className="mr2">
            <UserAvatar imageUrl="https://i.pravatar.cc/140" isOnline size="xlarge" />
        </div>
        <div className="mr2">
            <UserAvatar imageUrl="https://i.pravatar.cc/140" isOnline size="large" />
        </div>
        <div className="mr2">
            <UserAvatar imageUrl="https://i.pravatar.cc/140" isOnline size="medium" />
        </div>
        <div className="mr2">
            <UserAvatar imageUrl="https://i.pravatar.cc/140" isOnline size="small" />
        </div>
        <div className="mr2">
            <UserAvatar imageUrl="https://i.pravatar.cc/140" isOnline size="xsmall" />
        </div>
    </div>
</div>
```
