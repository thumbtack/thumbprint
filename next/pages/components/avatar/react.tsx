import React from 'react';
import type { InferGetStaticPropsType } from 'next';
import { EntityAvatar, UserAvatar } from '@thumbtack/thumbprint-react';
import { MDXComponentPage } from '../../../components/mdx/mdx';
import getComponentPageStaticProps from '../../../utils/get-component-page-static-props';
import { CodeExperimental, H2, H3, P } from '../../../components/mdx/components';
import DeprecatedComponentAlert from '../../../components/thumbprint-components/deprecated-component-alert/deprecated-component-alert';

export const getStaticProps = () => {
    return getComponentPageStaticProps({
        title: 'Avatar',
        description: 'Display user images and badges on Thumbtack.',
        componentId: 'avatar',
        platformId: 'react',
    });
};

export default function OverviewAbout({
    layoutProps,
    componentPageProps,
}: InferGetStaticPropsType<typeof getStaticProps>) {
    return (
        <MDXComponentPage componentPageProps={componentPageProps} layoutProps={layoutProps}>
            <H2>Avatar variations</H2>

            <P>
                Avatars are available as two components:{' '}
                <CodeExperimental>UserAvatar</CodeExperimental> and{' '}
                <CodeExperimental>EntityAvatar</CodeExperimental>.
            </P>

            <CodeExperimental language="jsx">
                <div className="flex">
                    <div className="mr3">
                        <EntityAvatar imageUrl="https://i.pravatar.cc/140" size="xlarge" />
                    </div>
                    <div>
                        <UserAvatar imageUrl="https://i.pravatar.cc/140" size="xlarge" />
                    </div>
                </div>
            </CodeExperimental>

            <P>
                <CodeExperimental>UserAvatar</CodeExperimental> is for people or users whereas{' '}
                <CodeExperimental>EntityAvatar</CodeExperimental> is for companies, businesses, or
                services.
            </P>

            <H2>Avatar sizes</H2>

            <P>
                Both <CodeExperimental>UserAvatar</CodeExperimental> and{' '}
                <CodeExperimental>EntityAvatar</CodeExperimental> are available in five sizes
                ranging from <CodeExperimental>xlarge</CodeExperimental> to{' '}
                <CodeExperimental>xsmall</CodeExperimental>.
            </P>

            <CodeExperimental language="jsx">
                <>
                    <div className="flex items-center mb4">
                        <div className="mr3">
                            <UserAvatar imageUrl="https://i.pravatar.cc/140" size="xlarge" />
                        </div>
                        <div className="mr3">
                            <UserAvatar imageUrl="https://i.pravatar.cc/140" size="large" />
                        </div>
                        <div className="mr3">
                            <UserAvatar imageUrl="https://i.pravatar.cc/140" size="medium" />
                        </div>
                        <div className="mr3">
                            <UserAvatar imageUrl="https://i.pravatar.cc/140" size="small" />
                        </div>
                        <div className="mr3">
                            <UserAvatar imageUrl="https://i.pravatar.cc/140" size="xsmall" />
                        </div>
                    </div>
                    <div className="flex items-center">
                        <div className="mr3">
                            <EntityAvatar imageUrl="https://i.pravatar.cc/140" size="xlarge" />
                        </div>
                        <div className="mr3">
                            <EntityAvatar imageUrl="https://i.pravatar.cc/140" size="large" />
                        </div>
                        <div className="mr3">
                            <EntityAvatar imageUrl="https://i.pravatar.cc/140" size="medium" />
                        </div>
                        <div className="mr3">
                            <EntityAvatar imageUrl="https://i.pravatar.cc/140" size="small" />
                        </div>
                        <div className="mr3">
                            <EntityAvatar imageUrl="https://i.pravatar.cc/140" size="xsmall" />
                        </div>
                    </div>
                </>
            </CodeExperimental>

            <H2>Avatars without images</H2>

            <P>
                Avatars without images can display the the user or entityâ€™s initials instead. The
                initials are assigned to a color based on the first letter in the{' '}
                <CodeExperimental>initials</CodeExperimental> prop.
            </P>

            <CodeExperimental language="jsx">
                <>
                    <div className="flex mb4">
                        <div className="mr2">
                            <UserAvatar initials="AA" size="large" />
                        </div>
                        <div className="mr2">
                            <UserAvatar initials="BA" size="large" />
                        </div>
                        <div className="mr2">
                            <UserAvatar initials="CA" size="large" />
                        </div>
                        <div className="mr2">
                            <UserAvatar initials="DA" size="large" />
                        </div>
                        <div className="mr2">
                            <UserAvatar initials="EA" size="large" />
                        </div>
                        <div>
                            <UserAvatar initials="HA" size="large" />
                        </div>
                    </div>
                    <div className="flex">
                        <div className="mr2">
                            <EntityAvatar initial="A" size="large" />
                        </div>
                        <div className="mr2">
                            <EntityAvatar initial="B" size="large" />
                        </div>
                        <div className="mr2">
                            <EntityAvatar initial="C" size="large" />
                        </div>
                        <div className="mr2">
                            <EntityAvatar initial="D" size="large" />
                        </div>
                        <div className="mr2">
                            <EntityAvatar initial="E" size="large" />
                        </div>
                        <div>
                            <EntityAvatar initial="F" size="large" />
                        </div>
                    </div>
                </>
            </CodeExperimental>

            <H2>Avatar badges</H2>

            <P>
                Badges can be added to avatars to denote information including hired status and
                online status.
            </P>

            <H3>Online</H3>

            <P>This badge indicates that a user or entity is online.</P>

            <CodeExperimental language="jsx">
                <div>
                    <div className="flex items-center mb4">
                        <div className="mr2">
                            <UserAvatar
                                imageUrl="https://i.pravatar.cc/140"
                                isOnline
                                size="xlarge"
                            />
                        </div>
                        <div className="mr2">
                            <UserAvatar
                                imageUrl="https://i.pravatar.cc/140"
                                isOnline
                                size="large"
                            />
                        </div>
                        <div className="mr2">
                            <UserAvatar
                                imageUrl="https://i.pravatar.cc/140"
                                isOnline
                                size="medium"
                            />
                        </div>
                        <div className="mr2">
                            <UserAvatar
                                imageUrl="https://i.pravatar.cc/140"
                                isOnline
                                size="small"
                            />
                        </div>
                        <div className="mr2">
                            <UserAvatar
                                imageUrl="https://i.pravatar.cc/140"
                                isOnline
                                size="xsmall"
                            />
                        </div>
                    </div>

                    <div className="flex items-center">
                        <div className="mr2">
                            <EntityAvatar
                                imageUrl="https://i.pravatar.cc/140"
                                isOnline
                                size="xlarge"
                            />
                        </div>
                        <div className="mr2">
                            <EntityAvatar
                                imageUrl="https://i.pravatar.cc/140"
                                isOnline
                                size="large"
                            />
                        </div>
                        <div className="mr2">
                            <EntityAvatar
                                imageUrl="https://i.pravatar.cc/140"
                                isOnline
                                size="medium"
                            />
                        </div>
                        <div className="mr2">
                            <EntityAvatar
                                imageUrl="https://i.pravatar.cc/140"
                                isOnline
                                size="small"
                            />
                        </div>
                        <div className="mr2">
                            <EntityAvatar
                                imageUrl="https://i.pravatar.cc/140"
                                isOnline
                                size="xsmall"
                            />
                        </div>
                    </div>
                </div>
            </CodeExperimental>

            <H3>Checked</H3>

            <DeprecatedComponentAlert>
                The checked badge is deprecated. To indicate hired status of a user, show this
                information separately next to the avatar. Only the online badge is supported by
                both components. Checked is only supported by UserAvatar.
            </DeprecatedComponentAlert>

            <CodeExperimental language="jsx">
                <div className="flex items-center mb4">
                    <div className="mr2">
                        <UserAvatar imageUrl="https://i.pravatar.cc/140" isChecked size="xlarge" />
                    </div>
                    <div className="mr2">
                        <UserAvatar imageUrl="https://i.pravatar.cc/140" isChecked size="large" />
                    </div>
                    <div className="mr2">
                        <UserAvatar imageUrl="https://i.pravatar.cc/140" isChecked size="medium" />
                    </div>
                    <div className="mr2">
                        <UserAvatar imageUrl="https://i.pravatar.cc/140" isChecked size="small" />
                    </div>
                    <div className="mr2">
                        <UserAvatar imageUrl="https://i.pravatar.cc/140" isChecked size="xsmall" />
                    </div>
                </div>
            </CodeExperimental>
        </MDXComponentPage>
    );
}
