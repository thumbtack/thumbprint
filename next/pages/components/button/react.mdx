import { MDXComponentPage } from '../../../components/mdx/mdx';
import baseGetStaticProps from '../../../utils/mdx-get-static-props';
export const getStaticProps = ctx => baseGetStaticProps(ctx, metadata);
export default props => <MDXComponentPage {...props} {...metadata} />;

export const metadata = {
    title: 'Button',
    description: 'Clickable elements used to perform actions.',
    component: {
        id: 'button',
        platformId: 'react',
    },
};

import { Button, TextButton } from '@thumbtack/thumbprint-react';
import DeprecatedComponentAlert from '../../../components/thumbprint-components/deprecated-component-alert/deprecated-component-alert.tsx';

## Button themes

These are the available `Button` styles for use on light background colors.

```jsx
<ButtonRow>
    <Button>Primary</Button>
    <Button theme="secondary">Secondary</Button>
    <Button theme="tertiary">Tertiary</Button>
    <Button theme="caution">Caution</Button>
</ButtonRow>
```

### Use on dark backgrounds

The `solid` theme works well on dark backgrounds.

```jsx theme=dark
<Button theme="solid">Solid</Button>
```

## Small buttons

Buttons are `large` by default. The `size` prop allows you to render a `small` button.

```jsx
<ButtonRow>
    <Button size="small">Primary</Button>
    <Button size="small" theme="secondary">
        Secondary
    </Button>
    <Button size="small" theme="tertiary">
        Tertiary
    </Button>
    <Button size="small" theme="caution">
        Caution
    </Button>
</ButtonRow>
```

## Disabled buttons

The `isDisabled` prop visually and functionally disables the button.

```jsx
<ButtonRow>
    <Button isDisabled theme="primary">
        Primary
    </Button>
    <Button isDisabled theme="secondary">
        Secondary
    </Button>
    <Button isDisabled theme="tertiary">
        Tertiary
    </Button>
    <Button isDisabled theme="caution">
        Caution
    </Button>
</ButtonRow>
```

```jsx theme=dark
<ButtonRow>
    <Button isDisabled theme="solid">
        Solid
    </Button>
</ButtonRow>
```

## Full width buttons

```jsx
<Button width="full">Send Quote</Button>
```

### Full width on small screens

This button becomes full width on viewports that are smaller than [our `small` breakpoint](/tokens/#section-breakpoint). The width is `auto` on larger screens.

```jsx
<Button width="full-below-small">Send Quote</Button>
```

## Icon within a button

The `icon` prop accepts any small icon from [Thumbprint Icons](/icons/).

This prop renders an icon left within button.

```jsx
<Button icon={<ContentModifierMessageSmall />}>Send Message</Button>
```

### Icon right within the button

`iconRight` renders an icon right within button.

```jsx
<Button iconRight={<ContentModifierMessageSmall />}>Send Message</Button>
```

## Button with loading indicator

These loading indicators should be used when buttons are processing. They are avialble in the `primary`, `secondary`, and `tertiary` button themes.

```jsx
<ButtonRow>
    <Button isLoading>Send Quote</Button>
    <Button theme="secondary" isLoading>
        Send Quote
    </Button>
    <Button theme="tertiary" isLoading>
        Send Quote
    </Button>
</ButtonRow>
```

## Buttons that look like links

The `TextButton` component renders button elements that visually appear as text links. These buttons accept an `onClick` and render as inline elements.

```jsx
<React.Fragment>
    The Pro exceeded our expectations and did a fantastic&hellip; <TextButton>Read more</TextButton>
</React.Fragment>
```

### Adding an icon with text

The `TextButton` component provides a `iconLeft` prop to help vertically position icons alongside text.

```jsx
<b>
    <TextButton iconLeft={<NavigationArrowLeftSmall />}>Back</TextButton>
</b>
```

### Adding an icon with text on the right

The `TextButton` component also allows an icon to the right of the text.

```jsx
<b>
    <TextButton iconRight={<NavigationArrowRightSmall />}>Forward</TextButton>
</b>
```

### Icon that inherits color of parent

This icon inherits the color of its parent with `theme="inherit"`.

```jsx theme=dark
<div style={{ color: tokens.tpColorWhite }}>
    <TextButton theme="inherit" iconLeft={<NavigationCloseMedium />} accessibilityLabel="Close" />
</div>
```

### Changing the font size and weight

These buttons inherit the font size and font weight of their container.

```jsx
<Text size={3}>
    The Pro exceeded our expectations and did a fantastic&hellip;{' '}
    <b>
        <TextButton>Read more</TextButton>
    </b>
</Text>
```
