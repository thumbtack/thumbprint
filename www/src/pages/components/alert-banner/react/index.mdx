---
title: Alert Banner
description: Important account information displayed at the top of a screen
---

import { graphql } from 'gatsby';
import { ComponentHeader, ComponentFooter } from 'components/thumbprint-components';

<ComponentHeader data={props.data} />

## Usage

`AlertBanner` should be used as a banner at the very top of the page.

Call-to-actions (e.g. `<a>` tags in the `children` prop) should be the last item in the `children`.

## Themes

### Caution

```jsx
<AlertBanner theme="caution">
    Your pre-paid credits are running low. Purchase new credits before you run out.{' '}
    <a href="https://example.com/">Purchase credits</a>
</AlertBanner>
```

### Info

```jsx
<AlertBanner theme="info">
    You havenâ€™t finish setting up. Add 3 reviews and turn on targeting.{' '}
    <a href="https://example.com/">Finish setup</a>
</AlertBanner>
```

### Warning

```jsx
<AlertBanner theme="warning">
    Oops. Your payment failed! Update your payment details to continue using Thumbtack.{' '}
    <a href="https://example.com/">Update payment</a>
</AlertBanner>
```

<ComponentFooter data={props.data} />

export const pageQuery = graphql`
    {
        # Get links to by path to display in the navbar.
        platformNav: allSitePage(filter: { path: { glob: "/components/alert-banner/*/" } }) {
            edges {
                node {
                    ...PlatformNavFragment
                }
            }
        }
        # Get package information by NPM package name.
        packageTable: thumbprintComponent(name: { eq: "@thumbtack/thumbprint-react" }) {
            ...PackageTableFragment
        }
        # Get component props by path to component file.
        reactComponentProps: allFile(
            filter: { relativePath: { in: ["thumbprint-react/components/AlertBanner/index.tsx"] } }
        ) {
            ...ReactComponentPropsFragment
        }
    }
`;
