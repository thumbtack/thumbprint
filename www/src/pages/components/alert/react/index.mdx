---
title: Alert
description: Vibrant alert messages used as page banners or within forms.
---

import { graphql } from 'gatsby';
import { ComponentHeader, ComponentFooter } from 'components/thumbprint-components';

<ComponentHeader data={props.data} />

## Banner alerts

These alerts should be used at the very top of a page.

```jsx
<BannerAlert theme="bad">
    <p>An Alert!</p>
</BannerAlert>
```

```jsx
<BannerAlert theme="good">
    <p>An Alert!</p>
</BannerAlert>
```

```jsx
<BannerAlert theme="warning">
    <p>A Warning!</p>
</BannerAlert>
```

```jsx
<BannerAlert theme="info">
    <p>An Alert!</p>
</BannerAlert>
```

## In-page alerts

These alerts should be used inside other page elements, e.g. forms.

```jsx
<InPageAlert theme="bad">
    <p>
        Your credit card is about to expire. To keep quoting on job requests please
        <a href="https://example.com/">update your credit card now</a>.
    </p>
</InPageAlert>
```

```jsx
<InPageAlert theme="good">
    <p>
        Your credit card is about to expire. To keep quoting on job requests please
        <a href="https://example.com/">update your credit card now</a>.
    </p>
</InPageAlert>
```

```jsx
<InPageAlert theme="warning">
    <p>
        Your credit card is about to expire. To keep quoting on job requests please
        <a href="https://example.com/">update your credit card now</a>.
    </p>
</InPageAlert>
```

```jsx
<InPageAlert theme="info">
    <p>
        Your credit card is about to expire. To keep quoting on job requests please
        <a href="https://example.com/">update your credit card now</a>.
    </p>
</InPageAlert>
```

<ComponentFooter data={props.data} />

export const pageQuery = graphql`
    {
        # Get links to by path to display in the navbar.
        platformNav: allSitePage(filter: { path: { glob: "/components/alert/*/" } }) {
            edges {
                node {
                    ...PlatformNavFragment
                }
            }
        }
        # Get package information by NPM package name.
        packageTable: thumbprintComponent(name: { eq: "@thumbtack/thumbprint-react" }) {
            ...PackageTableFragment
        }
        # Get component props by path to component file.
        reactComponentProps: file(
            relativePath: { eq: "thumbprint-react/components/Alert/index.tsx" }
        ) {
            ...ReactComponentPropsFragment
        }
    }
`;
