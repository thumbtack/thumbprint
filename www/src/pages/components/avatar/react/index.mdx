---
title: Avatar
description: Display user images and badges on Thumbtack.
---

import { graphql } from 'gatsby';
import {
    ComponentHeader,
    ComponentFooter,
    DeprecatedComponentAlert,
} from 'components/thumbprint-components';

<ComponentHeader data={props.data} />

## Avatar variations

Avatars are available as two components: `UserAvatar` and `EntityAvatar`.

```jsx
<div className="flex">
    <div className="mr3">
        <EntityAvatar imageUrl="https://www.placecage.com/640/480" size="xlarge" />
    </div>
    <div>
        <UserAvatar imageUrl="https://www.placecage.com/640/480" size="xlarge" />
    </div>
</div>
```

`UserAvatar` is for people or users whereas `EntityAvatar` is for companies, businesses, or services.

## Avatar sizes

Both `UserAvatar` and `EntityAvatar` are available in five sizes ranging from `xlarge` to `xsmall`.

```jsx
<>
    <div className="flex items-center mb4">
        <div className="mr3">
            <UserAvatar imageUrl="https://www.placecage.com/640/480" size="xlarge" />
        </div>
        <div className="mr3">
            <UserAvatar imageUrl="https://www.placecage.com/640/480" size="large" />
        </div>
        <div className="mr3">
            <UserAvatar imageUrl="https://www.placecage.com/640/480" size="medium" />
        </div>
        <div className="mr3">
            <UserAvatar imageUrl="https://www.placecage.com/640/480" size="small" />
        </div>
        <div className="mr3">
            <UserAvatar imageUrl="https://www.placecage.com/640/480" size="xsmall" />
        </div>
    </div>
    <div className="flex items-center">
        <div className="mr3">
            <EntityAvatar imageUrl="https://www.placecage.com/640/480" size="xlarge" />
        </div>
        <div className="mr3">
            <EntityAvatar imageUrl="https://www.placecage.com/640/480" size="large" />
        </div>
        <div className="mr3">
            <EntityAvatar imageUrl="https://www.placecage.com/640/480" size="medium" />
        </div>
        <div className="mr3">
            <EntityAvatar imageUrl="https://www.placecage.com/640/480" size="small" />
        </div>
        <div className="mr3">
            <EntityAvatar imageUrl="https://www.placecage.com/640/480" size="xsmall" />
        </div>
    </div>
</>
```

## Avatars without images

Avatars without images can display the the user or entityâ€™s initials instead. The initials are assigned to a color based on the first letter in the `initials` prop.

```jsx
<>
    <div className="flex mb4">
        <div className="mr2">
            <UserAvatar initials="AA" size="large" />
        </div>
        <div className="mr2">
            <UserAvatar initials="BA" size="large" />
        </div>
        <div className="mr2">
            <UserAvatar initials="CA" size="large" />
        </div>
        <div className="mr2">
            <UserAvatar initials="DA" size="large" />
        </div>
        <div className="mr2">
            <UserAvatar initials="EA" size="large" />
        </div>
        <div>
            <UserAvatar initials="HA" size="large" />
        </div>
    </div>
    <div className="flex">
        <div className="mr2">
            <EntityAvatar initial="A" size="large" />
        </div>
        <div className="mr2">
            <EntityAvatar initial="B" size="large" />
        </div>
        <div className="mr2">
            <EntityAvatar initial="C" size="large" />
        </div>
        <div className="mr2">
            <EntityAvatar initial="D" size="large" />
        </div>
        <div className="mr2">
            <EntityAvatar initial="E" size="large" />
        </div>
        <div>
            <EntityAvatar initial="F" size="large" />
        </div>
    </div>
</>
```

## Avatar badges

Badges can be added to avatars to denote information including hired status and online status.

### Online

This badge indicates that a user or entity is online.

```jsx
<div>
    <div className="flex items-center mb4">
        <div className="mr2">
            <UserAvatar imageUrl="https://www.placecage.com/640/480" isOnline size="xlarge" />
        </div>
        <div className="mr2">
            <UserAvatar imageUrl="https://www.placecage.com/640/480" isOnline size="large" />
        </div>
        <div className="mr2">
            <UserAvatar imageUrl="https://www.placecage.com/640/480" isOnline size="medium" />
        </div>
        <div className="mr2">
            <UserAvatar imageUrl="https://www.placecage.com/640/480" isOnline size="small" />
        </div>
        <div className="mr2">
            <UserAvatar imageUrl="https://www.placecage.com/640/480" isOnline size="xsmall" />
        </div>
    </div>

    <div className="flex items-center">
        <div className="mr2">
            <EntityAvatar imageUrl="https://www.placecage.com/640/480" isOnline size="xlarge" />
        </div>
        <div className="mr2">
            <EntityAvatar imageUrl="https://www.placecage.com/640/480" isOnline size="large" />
        </div>
        <div className="mr2">
            <EntityAvatar imageUrl="https://www.placecage.com/640/480" isOnline size="medium" />
        </div>
        <div className="mr2">
            <EntityAvatar imageUrl="https://www.placecage.com/640/480" isOnline size="small" />
        </div>
        <div className="mr2">
            <EntityAvatar imageUrl="https://www.placecage.com/640/480" isOnline size="xsmall" />
        </div>
    </div>
</div>
```

### Checked

<DeprecatedComponentAlert>
    The checked badge is deprecated. To indicate hired status of a user, show this information
    separately next to the avatar. Only the online badge is supported by both components. Checked is
    only supported by UserAvatar.
</DeprecatedComponentAlert>

```jsx
<div className="flex items-center mb4">
    <div className="mr2">
        <UserAvatar imageUrl="https://www.placecage.com/640/480" isChecked size="xlarge" />
    </div>
    <div className="mr2">
        <UserAvatar imageUrl="https://www.placecage.com/640/480" isChecked size="large" />
    </div>
    <div className="mr2">
        <UserAvatar imageUrl="https://www.placecage.com/640/480" isChecked size="medium" />
    </div>
    <div className="mr2">
        <UserAvatar imageUrl="https://www.placecage.com/640/480" isChecked size="small" />
    </div>
    <div className="mr2">
        <UserAvatar imageUrl="https://www.placecage.com/640/480" isChecked size="xsmall" />
    </div>
</div>
```

<ComponentFooter data={props.data} />

export const pageQuery = graphql`
    {
        # Get links to by path to display in the navbar.
        platformNav: allSitePage(filter: { path: { glob: "/components/avatar/*/" } }) {
            edges {
                node {
                    ...PlatformNavFragment
                }
            }
        }
        # Get package information by NPM package name.
        packageTable: thumbprintComponent(name: { eq: "@thumbtack/thumbprint-react" }) {
            ...PackageTableFragment
        }
        # Get component props by path to component file.
        reactComponentProps: file(
            relativePath: { eq: "thumbprint-react/components/Avatar/index.tsx" }
        ) {
            ...ReactComponentPropsFragment
        }
    }
`;
