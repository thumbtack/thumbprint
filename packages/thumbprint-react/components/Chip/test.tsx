import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import FilterChip from './index';

test('renders `text` passed in', () => {
    const wrapper = render(<FilterChip text="Beverages" onClick={jest.fn()} />);
    expect(screen.getByRole('button')).toHaveTextContent('Beverages');
    expect(wrapper.baseElement).toMatchSnapshot();
});

test('renders the unselected state by default', () => {
    const wrapper = render(<FilterChip text="Beverages" onClick={jest.fn()} />);
    expect(screen.getByRole('button', { pressed: false })).toBeTruthy();
    expect(wrapper.baseElement).toMatchSnapshot();
});

test('renders the selected state', () => {
    const wrapper = render(<FilterChip text="Beverages" isSelected onClick={jest.fn()} />);
    expect(screen.getByRole('button', { pressed: true })).toBeTruthy();
    expect(wrapper.baseElement).toMatchSnapshot();
});

test('renders a proper onClick callback', () => {
    let count = 0;
    const onClick = (): void => {
        count += 1;
    };
    const wrapper = render(<FilterChip text="Beverages" onClick={onClick} />);
    fireEvent.click(screen.getByRole('button'));
    expect(count).toEqual(1);
    expect(wrapper.baseElement).toMatchSnapshot();
});

type ExamplePropTypes = {
    text: string;
};

function ExamplePage({ text }: ExamplePropTypes): JSX.Element {
    const [isSelected, setIsSelected] = React.useState(false);
    const onClick = (): void => {
        setIsSelected(!isSelected);
    };
    return <FilterChip text={text} onClick={onClick} isSelected={isSelected} />;
}

test('renders the correct selected state when clicked', () => {
    const wrapper = render(<ExamplePage text="Beverages" />);
    expect(screen.getByRole('button', { pressed: false })).toBeTruthy();
    fireEvent.click(screen.getByRole('button'));
    const button = screen.getByRole('button', { pressed: true });
    expect(button).toBeTruthy();
    expect(button).toHaveTextContent('Beverages');
    expect(wrapper.baseElement).toMatchSnapshot();
});

test('renders the correct text when clicked', () => {
    let isSelected = false;
    const onClick = (): void => {
        isSelected = !isSelected;
    };
    const { rerender } = render(
        <FilterChip
            text={isSelected ? 'Remove Beverages' : 'Include Beverages'}
            onClick={onClick}
            isSelected={isSelected}
        />,
    );
    let button = screen.getByRole('button', { pressed: false });
    expect(button).toBeTruthy();
    expect(button).toHaveTextContent('Include Beverages');
    fireEvent.click(screen.getByRole('button'));
    rerender(
        <FilterChip
            text={isSelected ? 'Remove Beverages' : 'Include Beverages'}
            onClick={onClick}
            isSelected={isSelected}
        />,
    );
    button = screen.getByRole('button', { pressed: true });
    expect(button).toBeTruthy();
    expect(button).toHaveTextContent('Remove Beverages');
});
