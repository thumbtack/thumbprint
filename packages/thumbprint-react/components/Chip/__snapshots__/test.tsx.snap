// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders \`text\` passed in 1`] = `
<body>
  <div>
    <button
      aria-pressed="false"
      class="filterChip br-pill mr1 truncate no-underlin mb1 ph3 pv2 b filterChipNotSelected"
      type="button"
    >
      Beverages
    </button>
  </div>
</body>
`;

exports[`renders a proper onClick callback 1`] = `
<body>
  <div>
    <button
      aria-pressed="false"
      class="filterChip br-pill mr1 truncate no-underlin mb1 ph3 pv2 b filterChipNotSelected"
      type="button"
    >
      Beverages
    </button>
  </div>
</body>
`;

exports[`renders the correct selected state when clicked 1`] = `
<body>
  <div>
    <button
      aria-pressed="true"
      class="filterChip br-pill mr1 truncate no-underlin mb1 ph3 pv2 b filterChipSelected"
      type="button"
    >
      Beverages
    </button>
  </div>
</body>
`;

exports[`renders the selected state 1`] = `
<body>
  <div>
    <button
      aria-pressed="true"
      class="filterChip br-pill mr1 truncate no-underlin mb1 ph3 pv2 b filterChipSelected"
      type="button"
    >
      Beverages
    </button>
  </div>
</body>
`;

exports[`renders the unselected state by default 1`] = `
<body>
  <div>
    <button
      aria-pressed="false"
      class="filterChip br-pill mr1 truncate no-underlin mb1 ph3 pv2 b filterChipNotSelected"
      type="button"
    >
      Beverages
    </button>
  </div>
</body>
`;
